prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -129,9 +129,7 @@ fun Bundle.getObservations() = this.entry.map { it.resource }.filterIsInstance<O
 
 fun Bundle.getObservationsWithCondition(codes: List<String>): List<Observation> =
     if (codes.isEmpty()) {
-        // TODO: consider throwing IllegalArgumentException here while implementing
-        //  https://github.com/CDCgov/prime-reportstream/issues/12705
-        emptyList()
+        throw IllegalStateException("Invalid mapped condition filter")
```

Comment:
InvalidState is meant to be used when a method is invoked at a time when the environment or app is not in a correct state for the call.... IllegalArgument seems more appropriate here.

Reply:
Makes more sense to me 👍  (and past me looking at the comment xD)